votepy/algorithms/p_algorithm.py

```diff
@@ -0,0 +1,48 @@
+import random
+
+from votepy.ordinal_election import OrdinalElection
+from typing import Callable, Iterable
+
+from votepy.structure.structure import algo, BaseAlgorithm
+
+
+@algo(name='p_algorithm')
+class PAlgorithm(BaseAlgorithm):
+    def __init__(self):
+        """An algorithm that calculates the winning committee using a scoring function given in votepy/chamberlin_courant.py.
+        Algorithm based on https://doi.org/10.1016/j.artint.2015.01.003
+        """
+        super().__init__()
+    def simple_score(self, p, C):
+        if not p:
+            return -1
+        m = len(p)
+        for idx, vote in enumerate(p):
+            if vote in C:
+                return m - idx - 1
+
+    def simple_profile_score(self, P, C):
+        return sum([self.simple_score(p, C) for p in P])
+
+    def prepare(self, scoring_function: Callable[[Iterable[int], int, int], list[int]]):
```

Comment:
You should document `prepare` function. It may be used by other developers in others rules' implementations